src/reducers/game.py

```diff
@@ -15,46 +24,76 @@ def game(state=None, action=None):
     if action is None:
         return state
 
-    ty = action.get('type')
-    if ty == 'NEXT_TURN':
-        next_state = dict(**state)
-        next_state['turn'] = (state['turn'] + 1) % 3
+    action_type = action.get('type')
+    if action_type == 'GAME_NEXT_STEP':
+        if state['step'] == 3:
+            return state
+        next_state = deepcopy(state)
+        next_state['step'] = state['step'] + 1
         return next_state
 
-    if ty == 'NEXT_PHASE':
-        next_state = dict(**state)
-        next_state['phase'] = (state['phase'] + 1) % 4
+    if action_type == 'GAME_NEXT_PHASE':
+        next_state = deepcopy(state)
+        if state['phase'] == 5:
+            next_state['phase'] = 1
+            return next_state
+        next_state['phase'] = state['phase'] + 1
```

Comment:
state['phase'] % 5 +1 properly wraps it rather than the if statement.  Doesn't matter either way